### ISSUE TITLE
Support selecting also other layer types, not just markers

### ISSUE THREAD

**Issue by zakjan** [zakjan/leaflet-lasso#2]
Support selecting also other layer types, not just markers

Library that can be used for polygons: @turf/intersect

**Comment by ajhalls** [comment]
I am also hoping to be able to do this with either Circle or CircleMarker, it seems to be an issue with the getSelectedLayers() function. Within there there is a check to see if the layer is a "Marker" and if so calls 

var r = e.toGeoJSON().geometry; 
which in turn calls getLatLng().

Simply changing the o.Marker to o.Circle will get it to try the circles, but it fails to convert the layer. I didn't see what was different in the data structure for the Marker vs the Circle that it shouldn't have worked, but it doesn't.

**Comment by zakjan** [comment]
The issue could be that GeoJSON doesn't have an official Circle geometry. There is a draft https://github.com/geojson/geojson-spec/wiki/Proposal---Circles-and-Ellipses-Geoms, never implemented yet.

In order to use turf's capabilities, we can convert Leaflet Circle to GeoJSON Polygon using @turf/circle.

**Comment by matonga** [comment]
Changing:

`if (layer instanceof L.Marker) {`

To:

`if (layer instanceof L.Marker || layer instanceof L.CircleMarker) {`

Would actually make it for the CircleMarker case.

**Comment by zakjan** [comment]
It would work only for the center point, the plan here is to support also intersecting with the outer circle

**Comment by matonga** [comment]
That's right for Circle layers. But what about CircleMarker layers? Those have radius in pixels, which in my opinion would be equivalent to trying to match the Marker's image silohuette.

**Comment by zakjan** [comment]
Unfortunately, Marker image silouhette is also not matched currently. Only real geocoordinates

**Comment by answerquest** [comment]
@matonga is there an unminified version of the js that I can do your edit in, to make it select CircleMarker?

**Comment by answerquest** [comment]
@matonga no worries I got it to work with the minified js: towards the end of the file:

```
(function(e){if(e instanceof o.Marker )
```
becomes:
```
(function(e){if(e instanceof o.Marker || e instanceof o.CircleMarker )
```

**Comment by zakjan** [comment]
Implemented in 2.0.0
